### PR TITLE
 fix flaky upload badging

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: go/libkb/notify_router.go [hunk 1]
```diff
@@ -561,7 +561,7 @@ func (n *NotifyRouter) HandleFSSyncEvent(ctx context.Context, arg keybase1.FSPat
 				// A send of a `FSSyncActivity` RPC with the notification
 				(keybase1.NotifyFSClient{
 					Cli: rpc.NewClient(xp, NewContextifiedErrorUnwrapper(n.G()), nil),
-				}).FSSyncActivity(ctx, arg)
+				}).FSSyncActivity(context.Background(), arg)
```
[the remark this answers is not in the document]
Comment (jzila) [reply go/libkb/notify_router.go]: I wonder if this is a similar cause to plaintext notifications on Android. cc @mmaxim

[the remark this answers is not in the document]
Comment (jzila) [reply go/libkb/notify_router.go]: In the separate PR, let's try to also port over any debug tags that could be in the original context.

**Comment by songgao** [comment]
> In the separate PR, let's try to also port over any debug tags that could be in the original context.

I don't think `context` exposes that, unfortunately. It's just `.Value` method. We have ["delayed context"](https://github.com/keybase/kbfs/blob/master/libkbfs/delayed_cancellation.go), but I'm not sure it's worth here or not. Thoughts?

File: go/libkb/notify_router.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (songgao) [reply go/libkb/notify_router.go]: @jzila I went through all notifications in that file, and I think the only ones that use the `ctx` passed in are the KBFS ones. All other notifications just use `context.Background()` which disconnects the cancellation.

[the remark this answers is not in the document]
Comment (songgao) [reply go/libkb/notify_router.go]: That said, I don't really know how plaintext notification on Android gets pushed to FE ...

**Comment by songgao** [comment]
OK merging this! I created https://keybase.atlassian.net/browse/KBFS-3651 for tracking the debug tag thing.

**Pull request merged**